src/track/beats.cpp

```diff
@@ -1,21 +1,156 @@
 
 #include "track/beats.h"
 
+#include "track/beatutils.h"
+
 namespace mixxx {
 
-int Beats::numBeatsInRange(double dStartSample, double dEndSample) {
+namespace {
+const int kFrameSize = 2;
+
+bool BeatLessThan(const track::io::Beat& beat1, const track::io::Beat& beat2) {
+    return beat1.frame_position() < beat2.frame_position();
+}
+
+double framesToSamples(const int frames) {
+    return frames * kFrameSize;
+}
+
+} // namespace
+
+Beats::Beats(const Track* track, const QVector<double>& beats, SINT iSampleRate)
+        : Beats(track, iSampleRate) {
+    if (beats.size() > 0) {
+        createFromBeatVector(beats);
+    }
+}
+
+Beats::Beats(const Track* track, const QByteArray& byteArray, SINT iSampleRate)
+        : Beats(track, iSampleRate) {
+    readByteArray(byteArray);
```

Comment:
I think readByteArray is only used by this constructor? In that case we can get rid of the readByteArray function and move its implementation here. Same for the createFromBeatVector function.